package/api/src/controllers/m1/survey/mod.rs

```diff
@@ -0,0 +1,214 @@
+
+    let mut survey_responses_by_question: HashMap<QuestionId, HashMap<String, u64>> =
+        HashMap::new();
+    for (i, question) in survey.questions.into_iter().enumerate() {
+        questions.insert(i as QuestionId, question);
+    }
+    for quota in survey_answers.quotas.into_iter() {
+        let quota_id = quota.id.unwrap_or_default();
+        quotas.insert(quota_id as QuotaId, quota.into());
+    }
+    for item in survey_answers.response_array.into_iter() {
+        let quota_id = item.quota_id;
+        let responded_at =
+            convert_rfc3339_to_timestamp_millis(&item.responded_at).unwrap_or_default();
+        let mut responses: Vec<SurveyResultAnswerType> = vec![];
+
+        for (question_id, answer) in item.answers.into_iter().enumerate() {
+            let answer_type = if let Some(v) = answer.text_answer {
+                SurveyResultAnswerType::Text(v)
+            } else if let Some(v) = answer.choice_answer {
+                SurveyResultAnswerType::Select(v)
+            } else {
+                SurveyResultAnswerType::NotResponded
+            };
+            let text_list = match answer_type.clone() {
+                SurveyResultAnswerType::Text(v) => vec![v],
+                SurveyResultAnswerType::Select(v) => v,
+                _ => vec![],
+            };
+            for text in text_list {
+                survey_responses_by_question
+                    .entry(question_id as QuestionId)
+                    .or_insert_with(HashMap::new)
+                    .entry(text)
+                    .and_modify(|count| *count += 1)
+                    .or_insert(1);
+            }
+            responses.push(answer_type);
+        }
+        let result_answer: SurveyResultAnswer = SurveyResultAnswer {
+            responded_at,
+            quota_id,
+            responses,
+        };
+
+        answers.push(result_answer);
+    }
+
+    let doc_id = uuid::Uuid::new_v4().to_string();
+    let _ = db
+        .create(SurveyResultDocument {
+            gsi1: SurveyResultDocument::get_gsi1(&survey.creator, &survey.id),
+            id: doc_id.clone(),
+            user_id: survey.creator.clone(),
+            survey_id: survey.id.clone(),
+            r#type: SurveyResultDocument::get_type(),
+            created_at: chrono::Utc::now().timestamp_millis(),
+            expected_responses: survey.total_response_count.unwrap_or_default(),
+            actual_responses: survey.response_count.unwrap_or_default(),
+            answers,
+            quotas,
+            questions,
+            survey_responses_by_quota: response_count_map,
+            survey_responses_by_question,
+        })
+        .await;
+
+    Ok(doc_id)
+}
+
+async fn finalize_survey(
+    State(state): State<AxumState>,
+    Json(req): Json<AdminSurveyCompleteRequest>,
+) -> Result<Json<AdminSurveyCompleteResponse>, ApiError> {
+    // Get survey from DB
+    slog::debug!(root(), "finalize_survey Date : {}", req.ended_at);
```

Comment:
```suggestion
    let log = new_for_api(root(), "GET", "/v1/...").new(function => "finalize_survey");
    slog::debug!(log, ....)
```
이런식으로 로그를 남기면 나중에 로그를 찾기 좋을거에요.